From 26679d9344d892a97351d460442fefe1f1dcb33e Mon Sep 17 00:00:00 2001
From: Pekka Paalanen <pekka.paalanen@collabora.com>
Date: Thu, 4 Apr 2019 13:47:40 +0300
Subject: [PATCH 06/45] Rename matrix.h to libweston/matrix.h

matrix.h is a public installed header and even used by libweston.h.

See "Rename compositor.h to libweston/libweston.h" for rationale.

Signed-off-by: Pekka Paalanen <pekka.paalanen@collabora.com>
---
 clients/calibrator.c                   | 4 ++--
 clients/touch-calibrator.c             | 2 +-
 include/libweston/libweston.h          | 2 +-
 {shared => include/libweston}/matrix.h | 0
 include/libweston/meson.build          | 1 +
 libweston/meson.build                  | 1 -
 shared/matrix.c                        | 2 +-
 shared/meson.build                     | 1 +
 tests/matrix-test.c                    | 2 +-
 tests/meson.build                      | 2 +-
 10 files changed, 9 insertions(+), 8 deletions(-)
 rename {shared => include/libweston}/matrix.h (100%)

diff --git a/clients/calibrator.c b/clients/calibrator.c
index 778c23cf..2439fc94 100644
--- a/clients/calibrator.c
+++ b/clients/calibrator.c
@@ -38,7 +38,7 @@
 
 #include "window.h"
 #include "shared/helpers.h"
-#include "shared/matrix.h"
+#include <libweston/matrix.h>
 
 /* Our points for the calibration must be not be on a line */
 static const struct {
@@ -286,7 +286,7 @@ main(int argc, char *argv[])
 			break;
 		}
 	}
-	
+
 	display = display_create(&argc, argv);
 
 	if (display == NULL) {
diff --git a/clients/touch-calibrator.c b/clients/touch-calibrator.c
index 3c7e6ece..96f8c5a5 100644
--- a/clients/touch-calibrator.c
+++ b/clients/touch-calibrator.c
@@ -40,7 +40,7 @@
 
 #include "clients/window.h"
 #include "shared/helpers.h"
-#include "shared/matrix.h"
+#include <libweston/matrix.h>
 
 #include "weston-touch-calibration-client-protocol.h"
 
diff --git a/include/libweston/libweston.h b/include/libweston/libweston.h
index 16ad495b..e4ebc71d 100644
--- a/include/libweston/libweston.h
+++ b/include/libweston/libweston.h
@@ -41,7 +41,7 @@ extern "C" {
 #define WL_HIDE_DEPRECATED
 #include <wayland-server.h>
 
-#include "matrix.h"
+#include <libweston/matrix.h>
 #include "config-parser.h"
 #include "zalloc.h"
 #include <libweston/timeline-object.h>
diff --git a/shared/matrix.h b/include/libweston/matrix.h
similarity index 100%
rename from shared/matrix.h
rename to include/libweston/matrix.h
diff --git a/include/libweston/meson.build b/include/libweston/meson.build
index f6b55feb..e631e1d1 100644
--- a/include/libweston/meson.build
+++ b/include/libweston/meson.build
@@ -1,5 +1,6 @@
 install_headers(
 	'libweston.h',
+	'matrix.h',
 	'plugin-registry.h',
 	'timeline-object.h',
 	'windowed-output-api.h',
diff --git a/libweston/meson.build b/libweston/meson.build
index 0d6ec612..f508bd90 100644
--- a/libweston/meson.build
+++ b/libweston/meson.build
@@ -58,7 +58,6 @@ srcs_libweston = [
 
 install_headers(
 	'../shared/config-parser.h',
-	'../shared/matrix.h',
 	'../shared/zalloc.h',
 	subdir: dir_include_libweston
 )
diff --git a/shared/matrix.c b/shared/matrix.c
index 605db83c..4e8d6b40 100644
--- a/shared/matrix.c
+++ b/shared/matrix.c
@@ -37,7 +37,7 @@
 #include <wayland-server.h>
 #endif
 
-#include "matrix.h"
+#include <libweston/matrix.h>
 
 
 /*
diff --git a/shared/meson.build b/shared/meson.build
index 5b0d8d13..c3fd932a 100644
--- a/shared/meson.build
+++ b/shared/meson.build
@@ -17,6 +17,7 @@ lib_libshared = static_library(
 )
 dep_libshared = declare_dependency(
 	link_with: lib_libshared,
+	include_directories: public_inc,
 	dependencies: deps_libshared
 )
 
diff --git a/tests/matrix-test.c b/tests/matrix-test.c
index 520e12eb..03b92162 100644
--- a/tests/matrix-test.c
+++ b/tests/matrix-test.c
@@ -32,7 +32,7 @@
 #include <signal.h>
 #include <time.h>
 
-#include "shared/matrix.h"
+#include <libweston/matrix.h>
 
 struct inverse_matrix {
 	double LU[16];		/* column-major */
diff --git a/tests/meson.build b/tests/meson.build
index 03692f47..5274d8dd 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -97,7 +97,7 @@ dep_zucmain = declare_dependency(
 
 tests_standalone = [
 	['config-parser', [], [ dep_zucmain ]],
-	['matrix', [ '../shared/matrix.c' ], [ dep_libm ]],
+	['matrix', [ '../shared/matrix.c' ], [ dep_libm, dep_libshared.partial_dependency(includes: true) ]],
 	['string'],
 	[
 		'vertex-clip',
-- 
2.21.0

